src/common/local_file_system.cpp

```diff
@@ -52,32 +52,50 @@ static void AssertValidFileFlags(uint8_t flags) {
 }
 
 #ifndef _WIN32
-bool LocalFileSystem::FileExists(const string &filename) {
-	if (!filename.empty()) {
-		if (access(filename.c_str(), 0) == 0) {
-			struct stat status;
-			stat(filename.c_str(), &status);
-			if (S_ISREG(status.st_mode)) {
-				return true;
-			}
+
+static FileType GetFileTypeWithAccessCheck(const std::string &filename) {
+	if (filename.empty()) {
+		return FileType::FILE_TYPE_INVALID;
+	}
+	if (access(filename.c_str(), 0) != 0) {
+		if (errno == ENOENT) {
+			return FileType::FILE_TYPE_INVALID;
 		}
+		throw IOException("Cannot access file \"%s\": %s", filename, strerror(errno));
```

Comment:
Can we add a `TryGetFileType` method which generates the exception message on error? The `GetFileType` can then call that method and throw an exception on error.

Reply:
++

Reply:
qq: Did you want to expose TryGetFileType() on the public FileSystem interface?